src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -201,6 +201,27 @@ public static string Action(this IUrlHelper helper, string action, string contro
             });
         }
 
+        /// <summary>
+        /// Generates an absolute URL for the specified <paramref name="action"/> and action
+        /// <paramref name="values"/>, which contains the protocol (such as "http" or "https") and host name from the
+        /// current request.
+        /// </summary>
+        /// <param name="helper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="action">The name of the action that is used to generate URL.</param>
+        /// <param name="values">An object that contains action values.</param>
+        /// <returns>The generated absolute URL.</returns>
+        public static string AbsoluteAction(this IUrlHelper helper, string action, object values)
+        {
+            var httpContext = helper.ActionContext.HttpContext;
+            return helper.Action(new UrlActionContext()
+            {
+                Action = action,
+                Protocol = httpContext.Request.Scheme,
+                Host = httpContext.Request.Host.ToUriComponent(),
```

Comment:
Port?

Reply:
[Host](https://github.com/aspnet/HttpAbstractions/blob/master/src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs) includes the port